apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -509,6 +511,38 @@ object ConvertValueToDocumentReferencesForInQuery()
             }
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Query Where(Filter filter)
+        {
+            ValidateFilterForSentinels(filter);
+            var newFilter = (_filter == null) ? new Filter.CompositeFilter(new List<Filter>(), CompositeFilter.Types.Operator.And)
+            .WithAdditionalFilter(filter) : _filter.WithAdditionalFilter(filter);
+            return new Query(_root, _offset, _limit, _orderings, _projections, _startAt, _endAt, newFilter);
+        }
+
+        private void ValidateFilterForSentinels(Filter filter)
+        {
+            if (filter.GetType() == typeof(Filter.UnaryFilter))
+            {
+                Filter.UnaryFilter uFilter = (Filter.UnaryFilter) filter;
+                var convertedValue = ValueSerializer.Serialize(Database.SerializationContext, uFilter.Value);
+                ValidateNoSentinelsRecursively(convertedValue, "Sentinel values cannot be specified in filters");
+            }
+            else
+            {
+                Filter.CompositeFilter compositeFilter = (Filter.CompositeFilter) filter;
+                foreach(Filter.UnaryFilter f in compositeFilter.Filters)

```

Comment:
I'd expect a composite filter to be able to contain other composite filters.

Reply:
make sense this would be a recursive call then, missed that will update it in the final PR.